themes/default/content/blog/the-present-and-future-of-ai-and-iac/index.md

```diff
@@ -0,0 +1,111 @@
+
+In this article, we'll dig into both of these trends and the impacts that they are aleady having today, as well as the trends we are expecting to see (and build) for the near future.
+
+## AI transforming Infrastructure as Code
+
+### Today
+
+The scale of the cloud is truly incredible, and the sheer size of the AWS API is huge relative to almost any other computing platform, while AWS is just one of over 135 cloud providers Pulumi supports building and managing infrastructure for. The result is a total of more than 50,000 infrastructure resource types that can be managed, and over 1,000,000 configurable input properties on those resources. In practice, cloud engineers face a daunting task as they figure out how to combine these resources and properties to solve their unique problems. Finding an example or guidance for how to combine these together in the enormous variety of ways possible can be hard or impossible.  
+
+As a result, one of the most fundamental challenges in enabling developers to use the cloud is in helping them discover how to solve their problems using the enormous collection of available cloud infrastructure building blocks. For example, two years ago, I tried looking up how to **“deploy Metabase on Azure using managed PostreSQL and containers”** to solve a problem I was working on. I couldn’t find any content that directly answered this on the internet. Not just for Pulumi, there was no answer using any other IaC or deployment technology either.  
+
+![run Metabase in Azure](run-metabase-in-azure.png)
+
+AI is perfectly suited for this challenge. With [Pulumi AI](/ai/), I can ask a question and get back an answer to the exact things I’m looking for - even if that specific use case has never been answered before. Pulumi AI doesn’t always get it completely right on the first try, but it typically gets very close, and points me in the direction I’m looking for. When I try it’s suggestion and get errors, I can ask it to fix them up to iterate to a working solution.
+
+The adoption of Pulumi AI since its release less than a year ago has been incredible, with tens of thousands of users submitting hundreds of thousands of prompts. And, perhaps surprisingly, usage hasn’t been only or primarily driven by new Pulumi users. We’ve seen many of the most advanced Pulumi users with the deepest cloud knowledge leverage Pulumi AI the most heavily, helping them accelerate development of new infrastructure solutions, and discover more effective approaches to solving for their infrastructure needs using the best of their cloud platforms.
```

Comment:
```suggestion
The adoption of Pulumi AI since its release less than a year ago has been incredible, with tens of thousands of users submitting hundreds of thousands of prompts. And, perhaps surprisingly, usage hasn’t been driven only by or primarily by new Pulumi users. We’ve seen many of the most advanced Pulumi users with the deepest cloud knowledge leverage Pulumi AI the most heavily, helping them accelerate the development of new infrastructure solutions and discover more effective approaches to solving their infrastructure needs using the best of their cloud platforms.
```